### PR TITLE
Menudrop not receiving rtl from theme

### PR DESCRIPTION
**Pull request by IanKBovard**
#### What does this PR do?
Fixes issue #3755 by adding ThemeContext.Consumer to drop js

#### Where should the reviewer start?
Drop.js
#### What testing has been done on this PR?
Tested on storybook and using the codesandbox example here.
https://codesandbox.io/s/frosty-hofstadter-rhqkd

#### How should this be manually tested?
Storybook and codesanbox is best

#### Any background context you want to provide?
I'm assume this issue is occuring because of createPortal(). The drop container is created outside of root, and misses out on some props that are passed into grommet.

#### What are the relevant issues?
#3755 

#### Do the grommet docs need to be updated?
No

#### Should this PR be mentioned in the release notes?
No

#### Is this change backwards compatible or is it a breaking change?
Should be backwards compatible

### CODE REVIEW

**Comment by ShimiSun** [comment]
Is the suggested fix work similar to Layer? I think they should behave the same.

**Comment by IanKBovard** [comment]
ill double check

**Comment by IanKBovard** [comment]
it looks like rtl is applied to components within the layer. is that what you mean?

File: src/js/components/Drop/Drop.js [hunk 1]
```diff
@@ -46,12 +46,17 @@ const Drop = forwardRef(
 
     return dropContainer
       ? createPortal(
-          <DropContainer
-            ref={ref}
-            dropTarget={dropTarget}
-            restrictFocus={restrictFocus}
-            {...rest}
-          />,
+          <ThemeContext.Consumer>
```
Comment (ShimiSun) [inline src/js/components/Drop/Drop.js:49]: let's use 
```
const theme = useContext(ThemeContext) || defaultProps.theme;
```
instead

**Comment by IanKBovard** [comment]
Changes made

**Comment by ShimiSun** [comment]
linking to #3629.

**Pull request merged**